### ISSUE TITLE
Allow to specify source file instead of destination file for chezmoi apply

### ISSUE THREAD

**Issue by fwSmit** [twpayne/chezmoi#992]
When editing config files with vim I would like my changes being applied immediately when saving the file. I know of the command `chezmoi edit`, but that only applies changes after quitting the editor. The solution I use now is to run `chezmoi apply` from vim after saving, but that applies all files.
```vim
autocmd BufWritePost ~/.local/share/chezmoi/* ! chezmoi apply
```
I tried using 
```vim
autocmd BufWritePost ~/.local/share/chezmoi/* ! chezmoi apply %
```
where `%` is changed by vim to the current file. This doesn't work, however, since you have to specify the destination file, not the source file.
 
I propose adding a `--source` option to apply to apply the specified source file, so it can be used like this:
```vim
autocmd BufWritePost ~/.local/share/chezmoi/* ! chezmoi apply --source %
```

**Comment by twpayne** [comment]
Great idea! Implemented in https://github.com/twpayne/chezmoi/pull/990/commits/58894386065fb7a5e1913a375d4f95036c44146a.

Note that I had to use the `--source-path` option (instead of `--source`) as `--source` is already used to override the source directory.

**Comment by fwSmit** [comment]
> Great idea! Implemented in [5889438](https://github.com/twpayne/chezmoi/commit/58894386065fb7a5e1913a375d4f95036c44146a).

Thanks for implementing it so quickly! It works well. I did notice that you forgot to document it and add completions. Do you plan to do that in bulk?

**Comment by twpayne** [comment]
Yes, I'll do a bulk update of the documentation now that chezmoi2 is mostly stable.
